pytorch_lightning/accelerators/gpu.py

```diff
@@ -26,16 +26,19 @@
 class GPUAccelerator(Accelerator):
     """ Accelerator for GPU devices. """
 
+    def setup_environment(self) -> None:
+        super().setup_environment()
+        if "cuda" not in str(self.root_device):
+            raise MisconfigurationException(f"Device should be GPU, got {self.root_device} instead")
+        torch.cuda.set_device(self.root_device)
```

Comment:
reviewer: moving the call from `Accelerator.setup()` to here. Why? 

`setup_environment()` runs earlier in the `trainer._run()` and enables us to call broadcast etc. in a hook between, for example `Callback.setup()`

setting the devices is required for collective calls to work properly.